### ISSUE TITLE
add support for the workloadendpoint.active_instance_id field

### ISSUE THREAD

**Issue by rrayst** [projectcalico/libcalico#189]
The ```workloadendpoint.active_instance_id``` field was introduced in libcalico-go (https://github.com/projectcalico/libcalico-go/pull/396) to fix https://github.com/projectcalico/cni-plugin/issues/310 .
Currently, libcalico-go has the field, libcalico doesn't.

We should also add it to libcalico.

## Expected Behavior
Run calico-policy-controller and the cni-plugin e.g. on a three-node Kubernetes cluster. 
Delete a pod and immediately recreate it with the same name and schedule it on the same host (which is, what StatefulSets do when their pods vanish; scheduling to the same host can happen).
The newly created pod should have network connectivity.

## Current Behavior
The newly created pod looses network connectivity after ~30 seconds.

This is due to these inner workings:
When creating the new pod, Kuberenetes calls the cni-plugin with ```CNI_COMMAND=ADD``` and ```CNI_CONTAINERID=...the new pod's id...``` . The cni-plugin writes a workloadendpoint into etcd with the active_instance_id field set to the new pod's ID.
The calico-policy-controller updates the workloadendpoint (~after 2 seconds), effectively removing the active_instance_id field.
After ~30 seconds, the cni-plugin gets ```CNI_COMMAND=DEL``` call from Kubernetes with ```CNI_CONTAINERID=..the old pod's id...```. As the active_instance_id field is gone from etcd, the cni-plugin deletes the new pod's network connectivity.

To summarize: As calico-policy-controller removes the active_instance_id field from workloadendpoints when updating them, the effects described in https://github.com/projectcalico/cni-plugin/issues/310 reappear.

## Possible Solution
Adding the ```workloadendpoint.active_instance_id``` field to libcalico fixes this behavior.

**Comment by fasaxc** [comment]
@rrayst Thanks for the issue report.  FYI we have a golang version of the policy controller in the works so that should solve the problem.

**Comment by rrayst** [comment]
@fasaxc is that already public?

**Comment by fasaxc** [comment]
It is a work in progress on this (public) branch but I'm not sure if it's ready to try: https://github.com/projectcalico/k8s-policy/tree/golang

**Comment by caseydavenport** [comment]
@rrayst the golang code was merged to master today: https://github.com/projectcalico/k8s-policy

It's still hardening so there are likely to be bugs yet.   Feel free to try it out with the `calico/kube-policy-controller:master` tag

**Comment by timothysc** [comment]
There are a number of other issues that should probably be cross linked to this one.  

The behavioral manifestation is a crashlooping pod.  
https://github.com/projectcalico/calico/issues/1105 .   

FWIW a kubeadm installation running the most of the e2e suite would have exposed this issue.  

I'd highly recommend adding this to your process, and if you need/want help please feel free to reach out.  

/cc @jbeda

**Comment by caseydavenport** [comment]
> FWIW a kubeadm installation running the most of the e2e suite would have exposed this issue.

We do have a kubeadm rig which runs all of the Networking[Conformance] and NetworkPolicy e2e tests, but we don't see any issues in those tests.

@timothysc is there another set of tests which was exhibiting these failures?

**Comment by timothysc** [comment]
@caseydavenport That's a very small subset of the tests, I'd open up the focus to 'Conformance' and the skip to 'Alpha|Disruptive|Flaky'

**Comment by bcreane** [comment]
closing as fixed via k8s-policy master.
